tests/acceptance/generate-guard.spec.ts

```diff
@@ -1,80 +1,68 @@
-    });
-  });
-
-  afterEach(() => {
-    this.timeout(10000);
+  setupProject();
 
-    return tmp.teardown('./tmp');
-  });
-
-  it('ng generate guard my-guard', () => {
+  it('ng generate guard my-guard', (done) => {
     const appRoot = path.join(root, 'tmp/foo');
     const testPath = path.join(appRoot, 'src/app/my-guard.guard.ts');
     const testSpecPath = path.join(appRoot, 'src/app/my-guard.guard.spec.ts');
     const appModulePath = path.join(appRoot, 'src/app/app.module.ts');
 
     return ng(['generate', 'guard', 'my-guard'])
       .then(() => {
-        expect(fs.pathExistsSync(testPath)).to.equal(true);
-        expect(fs.pathExistsSync(testSpecPath)).to.equal(true);
+        expect(fs.pathExistsSync(testPath)).toBe(true);
+        expect(fs.pathExistsSync(testSpecPath)).toBe(true);
       })
       .then(() => fs.readFile(appModulePath, 'utf-8'))
       .then((content: string) => {
-        expect(content).not.to.matches(/import.*MyGuardGuard.*from '.\/my-guard.guard';/);
-        expect(content).not.to.matches(/providers:\s*\[MyGuardGuard\]/m);
+        expect(content).not.toMatch(/import.*MyGuardGuard.*from '.\/my-guard.guard';/);
+        expect(content).not.toMatch(/providers:\s*\[MyGuardGuard\]/m);
+        done();
       });
   });
 
-  it('ng generate guard my-guard --no-spec', () => {
+  it('ng generate guard my-guard --no-spec', (done) => {
     const appRoot = path.join(root, 'tmp/foo');
     const testPath = path.join(appRoot, 'src/app/my-guard.guard.ts');
     const testSpecPath = path.join(appRoot, 'src/app/my-guard.guard.spec.ts');
     const appModulePath = path.join(appRoot, 'src/app/app.module.ts');
 
     return ng(['generate', 'guard', 'my-guard', '--no-spec'])
       .then(() => {
-        expect(fs.pathExistsSync(testPath)).to.equal(true);
-        expect(fs.pathExistsSync(testSpecPath)).to.equal(false);
+        expect(fs.pathExistsSync(testPath)).toBe(true);
+        expect(fs.pathExistsSync(testSpecPath)).toBe(false);
       })
       .then(() => fs.readFile(appModulePath, 'utf-8'))
       .then((content: string) => {
-        expect(content).not.to.matches(/import.*MyGuardGuard.*from '.\/my-guard.guard';/);
-        expect(content).not.to.matches(/providers:\s*\[MyGuardGuard\]/m);
+        expect(content).not.toMatch(/import.*MyGuardGuard.*from '.\/my-guard.guard';/);
+        expect(content).not.toMatch(/providers:\s*\[MyGuardGuard\]/m);
+        done();
       });
   });
 
-  it('ng generate guard test' + path.sep + 'my-guard', () => {
+  it('ng generate guard test' + path.sep + 'my-guard', (done) => {
     fs.mkdirsSync(path.join(root, 'tmp', 'foo', 'src', 'app', 'test'));
     return ng(['generate', 'guard', 'test' + path.sep + 'my-guard']).then(() => {
       const testPath = path.join(root, 'tmp', 'foo', 'src', 'app', 'test', 'my-guard.guard.ts');
-      expect(fs.pathExistsSync(testPath)).to.equal(true);
+      expect(fs.pathExistsSync(testPath)).toBe(true);
+      done();
```

Comment:
Here and everywhere, you should `.then(done, done.fail)` to ensure you actually fail the test if something throws.